sonic_platform_base/module_base.py

```diff
@@ -196,6 +219,54 @@ def get_maximum_consumed_power(self):
         """
         raise NotImplementedError
 
+    ##############################################
+    # SmartSwitch methods
+    ##############################################
+
+    def get_dpu_id(self):
+        """
+        Retrieves the DPU ID. Returns None for non-smartswitch chassis.
+
+        Returns:
+            An integer, indicating the DPU ID. DPU0 returns 1, DPUX returns X+1
+        """
+        raise NotImplementedError
+
+    def get_reboot_cause(self):
+        """
+        Retrieves the cause of the previous reboot of the DPU module
+
+        Returns:
+            A tuple (string, string) where the first element is a string
+            containing the cause of the previous reboot. This string must
+            be one of the predefined strings in this class. If the first
+            string is "REBOOT_CAUSE_HARDWARE_OTHER", the second string can be
+            used to pass a description of the reboot cause.
+
+        """
+        raise NotImplementedError
+
+    def get_state_info(self):
```

Comment:
What is the difference between `get_state_info` and `get_oper_status`? Can you please add an example of the expected return value?

From the HLD, I understood that `get_oper_status` should return a simple online or offline state. And `get_state_info` should return the state progression. But from the description of the `get_oper_status`, it is different in this PR.

Reply:
Sure will add an example of the return value.